### DISCUSSION TITLE
Directus in micro-service architecture.

### DISCUSSION THREAD

**Discussion by u12206050** [directus/directus#14103] [Feature Requests]
Wouldn't it be awesome if you could use multiple Directus instances each concerned with its own Business Domain (micro-service style) and yet have them use data from each other using a combination of the autocomplete-api interface and good caching. 
ie:

Assume Directus instance A and B and C

- A has a list of people
- B has a list of companies
- C is a service for storing employee data

So in C, the company field stores the ID of companies found in service B, and the employee field is the ID of people found in service A.

**Further improvements:**
A custom display could be used with caching to retrieve the display values from those services
Api extended to act as a proxy towards other services. ie. Request to C could proxy requests further on to A and B

**Comment by benhaynes** [comment]
Absolutely! This is very similar to internal conversations we're having around multi-datastore support for Directus. Much of what you described is possible now, or could be easily implemented through custom extensions (that could be built into core).

However, whether using your microservice approach or our multi-datastore approach (with a single Directus instance), the most difficult challenge we're trying to solve is efficient JOINS across those databases. It's one thing to fetch a single item for an interface, but totally different to weave together relational data from 2+ places for a single `getItems` response.

Excited to keep this conversation going... solving this performantly will unlock a lot!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
